ballista/rust/scheduler/src/main.rs

```diff
@@ -100,10 +102,15 @@ async fn start_server(
             let scheduler_grpc_server =
                 SchedulerGrpcServer::new(scheduler_server.clone());
 
+            let flight_sql_server = FlightServiceServer::new(FlightSqlServiceImpl::new(
+                scheduler_server.clone(),
```

Comment:
Again, cloning the whole server seems wrong, but this PR isn't the first to do so. Is there something I'm missing here?

Reply:
Yeah, this probably needs to be cleaned up. The other places where we are cloning the scheduler server can probably be refactored to just take an `Arc<SchedulerState<>>`. Cloning the `EventLoop` might be more problematic though.